src/ScriptEditor/NetscriptDefinitions.d.ts

```diff
@@ -4092,6 +4093,362 @@ export interface Format {
+  isConnectedTo: boolean;
+  /** The amount of RAM in use (or blocked) on the server. */
+  ramUsed: number;
+  /** The total amount of RAM on the server. */
+  maxRam: number;
+  /** The amount of RAM allocated to the owner of the server. */
+  ownerAllocatedRam: number;
+  /** Whether the server has a backdoor installed. */
+  backdoorInstalled: boolean;
+  /** The amount of money available on the server. */
+  moneyAvailable: number;
+  /** The maximum amount of money on the server. */
+  moneyMax: number;
+  /** The cha level required to use heartBleed() on the server. */
+  charismaLevel: number;
+  /** The depth into the darknet of the server. */
+  depth: number;
+  /** The model ID of the server. Similar models have similar vulnerabilities */
+  modelId: string;
+};
+
+/**
+ * Details about a server's authentication schema
+ * @public
+ */
+export type ServerAuthDetails = {
+  /** True if the server is still online. */
+  isOnline: boolean;
+  /** True if the server is directly connected to the current server */
+  isConnectedToCurrentServer: boolean;
+  /** True if the current script has authenticated to this server with the right password using authenticate() or connectToSesssion() */
+  hasSession: boolean;
+  /** The model ID of the server. Similar models share vulnerabilities. */
+  modelId: string;
+  /** Static password reminder text set for this server. */
+  passwordHint: string;
+  /** Data from the passwordHint, if any. */
+  data: string;
+  /** The frequency (in seconds) of the server adding its own messages to its logs, visible with heartBleed(). */
+  logTrafficInterval: number;
+  /** The number of characters in the password */
+  passwordLength: number;
+  /** The character set used in the password */
+  passwordFormat: "numeric" | "alphabetic" | "alphanumeric" | "ASCII" | "unicode";
+};
+
+/**
+ * Response from attempting to scrape logs from a server
+ * @public
+ */
+export type HeartbleedOptions = {
+  /**
+   * If true, looks at the most recent log line but does not extract it. (Overrides logsToCapture.)
+   */
+  peek?: boolean;
+  /**
+   * The number of log lines to remove from the server, up to a max of 8. Default is 1.
+   */
+  logsToCapture?: number;
+  /**
+   * The number of additional milliseconds to add to the run time of the heartbleed request. Default is 0.
+   */
+  additionalMsec?: number;
+};
+
+/**
+ * Darknet API
+ * @remarks
+ * If you are not in BitNode-15, then you must have Source-File 15 in order to use this API.
+ * @public
+ */
+export interface Darknet {
+  /**
+   * Sends a network request to try to authenticate on a darkweb server. The target server must be directly connected
+   * to the server that the script is running on. The speed of authentication scales with the number of threads used.
+   *
+   * If successful, grants the script a session, allowing it to exec() scripts on that server, or scp() files from it.
```

Comment:
If what I am saying is accurate, then for clarity:
```suggestion
   * If successful, grants the script a session, allowing it to exec() scripts on that server, or scp() files from it. (scp() *to* the server is always allowed.)
```